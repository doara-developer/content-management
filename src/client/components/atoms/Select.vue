<template>
    <select class="app-select" v-model="selectValue">
        <option
            v-for="option in options"
            :key="option.value"
            :value="option.value"
        >
            {{ option.label }}
        </option>
    </select>
</template>
<script lang="ts">
import Vue, { PropType } from "vue";
import { OptionType } from "@common/types";

export default Vue.extend({
    props: {
        options: {
            type: Array as PropType<OptionType[]>,
            default: () => [],
        },
        value: {
            type: String,
            default: "",
        },
    },
    computed: {
        selectValue: {
            get(): string {
                return this.value;
            },
            set(value: string) {
                console.log(value);
                this.$emit("change", value);
            },
        },
    },
});
</script>
