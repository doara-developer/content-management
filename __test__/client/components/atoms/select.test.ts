import { shallowMount } from "@vue/test-utils";
import Select from "@client/components/atoms/Select.vue";

describe("Select.vue", () => {
    test("renders props.msg when passed", () => {
        const wrapper = shallowMount(Select);
        expect(wrapper).toMatchSnapshot();
    });
});
