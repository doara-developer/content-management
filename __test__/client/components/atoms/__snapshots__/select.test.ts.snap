// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select.vue renders props.msg when passed 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
